slick/src/main/scala/slick/jdbc/JdbcActionComponent.scala

```diff
@@ -652,7 +652,7 @@ trait JdbcActionComponent extends SqlActionComponent { self: JdbcProfile =>
       ResultSetInvoker[QR](_ => st.getGeneratedKeys)(pr => keyConverter.read(pr.rs).asInstanceOf[QR])
 
     // Returning keys from batch inserts is generally not supported
-    override protected def useBatchUpdates(implicit session: Backend#Session) = false
+    override protected def useBatchUpdates(implicit session: Backend#Session) = session.capabilities.supportsBatchUpdates
```

Comment:
I am not sure if this is actually correct. This code is part of `ReturningInsertActionComposerImpl`. Which means (actually I'm guessing based on the name) that a "returning" action is being used. This is normally used to return the generated value of a generated column (e.g. a generated primary key).

The comment `// Returning keys from batch inserts is generally not supported` suggests that most databases do not support this, even if they do support batch updates

Reply:
You state
> This would fix the issue and let us do batch inserts with returning ids on a database that supports it.

Perhaps we could add a separate capability to indicate which databases can return ids when batch inserts are used

Reply:
@WellingR I am working with MariaDB and that seems to work with returning IDs. Without this fix, I had to come up with my custom profile extending MySQLProfile and overriding the `createReturningInsertActionComposer` to achieve this at the moment.

I agree with you if you feel like only the supported database profiles should be updated, then I can come up with a PR just for MariaDB.